core/src/main/java/com/netflix/iceberg/BaseFileScanTask.java

```diff
@@ -74,4 +87,74 @@ public String toString() {
         .add("residual", residual())
         .toString();
   }
+
+  private static final class SplitScanTaskIterator implements Iterator<FileScanTask> {
```

Comment:
I'd like to see a unit test for this class that has a few example split cases, like file length = 100 and split size = 15 produces [0, 15), [15, 30), [30, 45), etc. That ensures there are no gaps between splits. Just reading data files doesn't make that guarantee.